software/controller/lib/hal/circular_buffer.h

```diff
@@ -69,17 +69,17 @@ template <class T, uint N> class CircularBuffer {
   // Returns false if the buffer is full.
   [[nodiscard]] bool Put(T dat) {
     BlockInterrupts block;
-    int h = head_ + 1;
-    if (h > N) {
-      h = 0;
+    int new_head = head_ + 1;
```

Comment:
this could even be `auto` to avoid future casting surprises?

Reply:
sure, why not?